pkg/snapshot/snapshotter/snapshotter_test.go

```diff
@@ -288,48 +288,108 @@ var _ = Describe("Snapshotter", func() {
+					Context("with snapshotter starting without first full snapshot", func() {
+						It("first snapshot should be a delta snapshot", func() {
+							var (
+								errCh           = make(chan error)
+								populatorStopCh = make(chan bool)
+								ssrStopCh       = make(chan struct{})
+							)
+							store, err = snapstore.GetSnapstore(&snapstore.Config{Container: path.Join(outputDir, "snapshotter_5.bkp")})
+							Expect(err).ShouldNot(HaveOccurred())
+							tlsConfig := etcdutil.NewTLSConfig(
+								certFile,
+								keyFile,
+								caFile,
+								insecureTransport,
+								insecureSkipVerify,
+								endpoints,
+								etcdUsername,
+								etcdPassword)
+							snapshotterConfig, err := NewSnapshotterConfig(
+								schedule,
+								store,
+								maxBackups,
+								deltaSnapshotIntervalSeconds,
+								DefaultDeltaSnapMemoryLimit,
+								etcdConnectionTimeout,
+								garbageCollectionPeriodSeconds,
+								GarbageCollectionPolicyExponential,
+								tlsConfig)
+							Expect(err).ShouldNot(HaveOccurred())
+
+							ssr = NewSnapshotter(
+								logger,
+								snapshotterConfig)
+
+							wg := &sync.WaitGroup{}
+							wg.Add(1)
+							// populating etcd so that snapshots will be taken
+							go populateEtcd(wg, logger, endpoints, errCh, populatorStopCh)
+
+							go func() {
+								<-time.After(testTimeout)
+								close(populatorStopCh)
+								wg.Wait()
+								close(ssrStopCh)
+							}()
+
+							err = ssr.Run(ssrStopCh, false)
+							Expect(err).ShouldNot(HaveOccurred())
+							list, err := store.List()
+							Expect(err).ShouldNot(HaveOccurred())
+							Expect(len(list)).ShouldNot(BeZero())
+							Expect(list[0].Kind).Should(Equal(snapstore.SnapshotKindDelta))
+						})
+					})
 
-						go func() {
-							<-time.After(testTimeout)
-							close(stopCh)
-						}()
-						err = ssr.Run(stopCh, true)
-						Expect(err).ShouldNot(HaveOccurred())
-						list, err := store.List()
-						Expect(err).ShouldNot(HaveOccurred())
-						Expect(len(list)).ShouldNot(BeZero())
+					Context("with snapshotter starting with full snapshot", func() {
+						It("should take periodic backups", func() {
+							stopCh := make(chan struct{})
+							store, err = snapstore.GetSnapstore(&snapstore.Config{Container: path.Join(outputDir, "snapshotter_6.bkp")})
+							Expect(err).ShouldNot(HaveOccurred())
+							tlsConfig := etcdutil.NewTLSConfig(
+								certFile,
+								keyFile,
+								caFile,
+								insecureTransport,
+								insecureSkipVerify,
+								endpoints,
+								etcdUsername,
+								etcdPassword)
+							snapshotterConfig, err := NewSnapshotterConfig(
+								schedule,
+								store,
+								maxBackups,
+								deltaSnapshotIntervalSeconds,
+								DefaultDeltaSnapMemoryLimit,
+								etcdConnectionTimeout,
+								garbageCollectionPeriodSeconds,
+								GarbageCollectionPolicyExponential,
+								tlsConfig)
+							Expect(err).ShouldNot(HaveOccurred())
+
```

Comment:
```suggestion
						wg := &sync.WaitGroup{}
							wg.Add(1)
							// populating etcd so that snapshots will be taken
							go populateEtcd(wg, logger, endpoints, errCh, populatorStopCh)
```